### PR TITLE
skip unset values in restricted_audience

### PR DESCRIPTION
**Pull request by rodriguez-facundo**

### CODE REVIEW

File: basemodels/pydantic/manifest/restricted_audience.py [hunk 1]
```diff
@@ -32,6 +32,14 @@ class RestrictedAudience(BaseModel):
...
     launch_group_id: Optional[conint(ge=0, strict=True)]
 
+    def dict(self, **kwargs):
+        kwargs["exclude_unset"] = kwargs.get("exclude_unset", True)
+        return super().dict(**kwargs)
+
+    def json(self, **kwargs):
+        kwargs["exclude_unset"] = kwargs.get("exclude_unset", True)
+        return super().json(**kwargs)
+
```
Comment (brdlyptrs) [inline basemodels/pydantic/manifest/restricted_audience.py:42]: Whats the use case to have two?

Comment (rodriguez-facundo) [reply basemodels/pydantic/manifest/restricted_audience.py:42]: dict is for Pydantic -> dict
json is for Pydantic -> str

Comment (rodriguez-facundo) [reply basemodels/pydantic/manifest/restricted_audience.py:42]: More than anything is to avoid getting all the fields that have value == None

**Review by brdlyptrs — APPROVED** [review]

File: basemodels/pydantic/manifest/restricted_audience.py (hunk 1 above)
Comment (gaieges) [reply basemodels/pydantic/manifest/restricted_audience.py]: Why not let the user specify that?

**Review by e271828- — APPROVED** [review]
It's getting pretty ugly to maintain support for two schema libraries here..

File: basemodels/pydantic/manifest/restricted_audience.py (hunk 1 above)
Comment (rodriguez-facundo) [reply basemodels/pydantic/manifest/restricted_audience.py]: the problem is that we have to remember to do `m.dict(exclude_unset=True)` everywhere. Also, for now, I just want to remove `None` fields from restricted_audience. But I can get to select "exclude_unset" only for restricted_audience. it propagates to the entire model. Pydantic propagates exclude_unset to all submodels https://github.com/samuelcolvin/pydantic/blob/a4cb4ee3bc57911a18b13643c44ab14e2ea28d33/pydantic/main.py#L894-L905

Comment (gaieges) [reply basemodels/pydantic/manifest/restricted_audience.py]: Why do you need `exclude_unset` on everything, is what I'm wondering?

**Pull request merged**